pubsub/mqclient/mq_client.go

```diff
@@ -0,0 +1,182 @@
+package mqclient
+
+import (
+	"context"
+	"errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/streadway/amqp"
+	"github.com/trustwallet/blockatlas/pubsub"
+	"go.uber.org/atomic"
+	"time"
+)
+
+var (
+	reconnectDelay = 5 * time.Second
+	resendDelay    = 5 * time.Second
+)
+
+type Client struct {
+	uri           string
+	conn          *amqp.Connection
+	channel       *amqp.Channel
+	prefetchCount int
+	streams       []*pubsub.Stream
+	ctx           context.Context
+	notifyClose   chan *amqp.Error
+	notifyConfirm chan amqp.Confirmation
+	isConnected   *atomic.Bool
+	alive         *atomic.Bool
+}
+
+func New(uri string, prefetchCount int, ctx context.Context) (client pubsub.Client) {
+	client = &Client{
+		uri:           uri,
+		ctx:           ctx,
+		alive:         atomic.NewBool(false),
+		isConnected:   atomic.NewBool(false),
+		prefetchCount: prefetchCount,
+		streams:       []*pubsub.Stream{},
+	}
+	return client
+}
+
+func (c *Client) Connect() error {
+	conn, err := amqp.Dial(c.uri)
+	if err != nil {
+		log.Error("Client.connect MQ Dial issue " + err.Error())
+		return err
+	}
+	ch, err := conn.Channel()
+	if err != nil {
+		log.Error("Client.connect MQ Channel issue " + err.Error())
```

Comment:
same log as below: `log.Error("Client.connect MQ Channel issue " + err.Error())`. Can you give unique names?